subprojects/dependency-management/src/main/java/org/gradle/api/internal/artifacts/ivyservice/ivyresolve/verification/writer/WriteDependencyVerificationFile.java

```diff
@@ -162,8 +163,10 @@ private void assertPgpHasChecksumFallback(List<String> kinds) {
     }
 
     private void warnAboutInsecureChecksums() {
-        if (checksums.stream().noneMatch(SECURE_CHECKSUMS::contains)) {
-            LOGGER.warn("You chose to generate " + String.join(" and ", checksums) + " checksums but they are all considered insecure. You should consider adding at least one of " + String.join(" or ", SECURE_CHECKSUMS) + ".");
+        if (isWriteVerificationFile()) {
```

Comment:
Maybe we could just do in validateChecksums, something like:
```
if (isWriteVerificationFile()) {
        assertSupportedChecksums();
        warnAboutInsecureChecksums();
}
```
instead of we handle it separately for every case. What do you think?